internal/etos/api/logarea.go

```diff
@@ -0,0 +1,241 @@
+		if !apierrors.IsNotFound(err) {
+			return serviceaccount, err
+		}
+		if err := r.Create(ctx, target); err != nil {
+			return target, err
+		}
+		return target, nil
+	}
+	return serviceaccount, r.Patch(ctx, target, client.StrategicMergeFrom(serviceaccount))
+}
+
+// reconcileService will reconcile the ETOS logarea service to its expected state.
+func (r *ETOSLogAreaDeployment) reconcileService(ctx context.Context, name types.NamespacedName, owner metav1.Object) (*corev1.Service, error) {
+	target := r.service(name)
+	if err := ctrl.SetControllerReference(owner, target, r.Scheme); err != nil {
+		return target, err
+	}
+	service := &corev1.Service{}
+	if err := r.Get(ctx, name, service); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return service, err
+		}
+		if err := r.Create(ctx, target); err != nil {
+			return target, err
+		}
+		return target, nil
+	}
+	return service, r.Patch(ctx, target, client.StrategicMergeFrom(service))
+}
+
+// serviceaccount creates a service account resource definition for the ETOS logarea.
+func (r *ETOSLogAreaDeployment) serviceaccount(name types.NamespacedName) *corev1.ServiceAccount {
+	return &corev1.ServiceAccount{
+		ObjectMeta: r.meta(name),
+	}
+}
+
+// deployment creates a deployment resource definition for the ETOS logarea.
+func (r *ETOSLogAreaDeployment) deployment(name types.NamespacedName) *appsv1.Deployment {
+	return &appsv1.Deployment{
+		ObjectMeta: r.meta(name),
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app.kubernetes.io/name":      name.Name,
+					"app.kubernetes.io/part-of":   "etos",
+					"app.kubernetes.io/component": "logarea",
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: r.meta(name),
+				Spec: corev1.PodSpec{
+					ServiceAccountName: name.Name,
+					Containers:         []corev1.Container{r.container(name)},
+				},
+			},
+		},
+	}
+}
+
+// service creates a service resource definition for the ETOS logarea.
+func (r *ETOSLogAreaDeployment) service(name types.NamespacedName) *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: r.meta(name),
+		Spec: corev1.ServiceSpec{
+			Ports: r.ports(),
+			Selector: map[string]string{
+				"app.kubernetes.io/name":      name.Name,
+				"app.kubernetes.io/part-of":   "etos",
+				"app.kubernetes.io/component": "logarea",
+			},
+		},
+	}
+}
+
+// container creates the container resource for the ETOS logarea deployment.
+func (r *ETOSLogAreaDeployment) container(name types.NamespacedName) corev1.Container {
+	probe := &corev1.Probe{
+		ProbeHandler: corev1.ProbeHandler{
+			HTTPGet: &corev1.HTTPGetAction{
+				Path:   "/logarea/v1alpha/selftest/ping",
+				Port:   intstr.FromString("http"),
+				Scheme: "HTTP",
+			},
+		},
+		TimeoutSeconds:   1,
+		PeriodSeconds:    10,
+		SuccessThreshold: 1,
+		FailureThreshold: 3,
+	}
+	return corev1.Container{
+		Name:  name.Name,
+		Image: r.Image,
```

Comment:
Applies to all services: Should this include ImagePullPolicy setting? Partly, if we use mutable tags sometimes ("latest") or if we want to update images more easily (like in a dev environment).

Reply:
Added to services where we allow image to be changed.